pyspelling/__init__.py

```diff
@@ -196,6 +198,10 @@ def _walk_src(self, targets, flags, pipeline):
                     except Exception as e:
                         err = self.get_error(e)
                         yield [filters.SourceText('', f, '', '', err)]
+            if not found_something:
+                raise ValueError(
+                    'The source target "{}" from the configuration did not match any files.'
```

Comment:
Did you mean to add the target pattern to the error? If not, you should remove `"{}"`, and if so, you should add the pattern to the output message.

Reply:
Actually, it probably makes sense to not include the pattern as we don't care if a single pattern finds no files. We only care if **all** the patterns find no files.

Reply:
Yeah, we should move this `ValueError` outside of the `for target in targets` loop. The fact that you can have multiple sources is just for convenience really. To some it makes more sense to just have a list of patterns, but there are multiple ways you can define multiple patterns. There is no intention to have inherent meaning behind using one vs the other:

1. list

    ```
    - '*.ext'
    - '*.ext2'
    ```

2. Using wcmatch's brace expansion

    ```
    -  '*.{ext,ext2}'
    ```

3. Using wcmatch's pattern delemiter (`|`)
    
    ```
    - '*.ext|*.ext2'
    ```

These are meant to be equivalent.

Reply:
I guess I'm open to a discussion about this though. Adding significance to each source entry when using the list methodology would definitely be **feature** vs **bug** and would require documentation if we decided to go down this road. Let me know your thoughts.

Reply:
Yes - Thanks for the great feedback - Yes I agree - I will adjust to not fail if one source does not match only if all of them fail, this does expose some risk of not picking up if one directory in a set does not match but like you point out you could be a victim of this even with a single source match that uses an if you use an expression with a disjunction. I'll fix it to display all the source targets if none of them match. Thanks for the pick-up.